### DISCUSSION TITLE
How to correctly generate UUIDs?

### DISCUSSION THREAD

**Discussion by malice00** [DependencyTrack/dependency-track#2505] [Q&A]
I'm working on some changes in the model and have added a new class. This class has a UUID-field which needs to be filled during the migration of data from the old model into the new.

What would be the correct way to go about this? Can I simply add a 'random_uuid()' (or equivalent) in my query, or is there some other way to set this value? Maybe I can somehow get a hold of DataNucleus' generator and have it generate values?

Any advice is greatly appreciated, I'm kind of stuck on this at the moment...

**Comment by stevespringett** [comment]
Datanucleus will automatically assign a UUID for any persistable object. It does this using the `customValueStrategy`. 

For example:
https://github.com/DependencyTrack/dependency-track/blob/master/src/main/java/org/dependencytrack/model/Component.java#L344

If the data migration is kicked off prior to the DN factories being invoked, then you'll need to this manually by calling:

```java
myNewObject.setUuid(UUID.randomUUID());
```
